### PR TITLE
Fix transition extensible object array bounds crash

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
Proposed fix to address #6287 .  Not sure if this is a general solution.  Don't know why the +20's were there.  Maybe they are necessary for other files?

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: src/Transition/InputProcessor.f90 [hunk 1]
```diff
@@ -408,10 +408,10 @@ INTEGER, EXTERNAL :: FindUnitNumber    ! External function that will get and ope
...
        WRITE(EchoInputFile,*) ' Number of Section Definitions=',NumSectionDefs
      ENDIF
-     IF (.not. ALLOCATED(LineItem%Numbers)) ALLOCATE (LineItem%Numbers(MaxNumericArgsFound+20))
-     IF (.not. ALLOCATED(LineItem%NumBlank)) ALLOCATE (LineItem%NumBlank(MaxNumericArgsFound+20))
-     IF (.not. ALLOCATED(LineItem%Alphas)) ALLOCATE (LineItem%Alphas(MaxAlphaArgsFound+20))
-     IF (.not. ALLOCATED(LineItem%AlphBlank)) ALLOCATE (LineItem%AlphBlank(MaxAlphaArgsFound+20))
+     IF (.not. ALLOCATED(LineItem%Numbers)) ALLOCATE (LineItem%Numbers(MaxNumericArgsFound))
+     IF (.not. ALLOCATED(LineItem%NumBlank)) ALLOCATE (LineItem%NumBlank(MaxNumericArgsFound))
+     IF (.not. ALLOCATED(LineItem%Alphas)) ALLOCATE (LineItem%Alphas(MaxAlphaArgsFound))
+     IF (.not. ALLOCATED(LineItem%AlphBlank)) ALLOCATE (LineItem%AlphBlank(MaxAlphaArgsFound))
```
Comment (rraustad) [inline src/Transition/InputProcessor.f90:414]: So why would having more elements in an array than needed cause a crash?

**Comment by rraustad** [comment]
I can't find the transition program to test the defect file used for #6200.

**Comment by rraustad** [comment]
@Myoldmopar what happens here when these are not the same size?

    SUBROUTINE ExtendObjectDefinition(ObjectNum,NumNewArgsLimit,ObjectDef)
        line 3112
        TempCharacter(1:ObjectDef(ObjectNum)%NumAlpha)=LineItem%Alphas

There's also some comment lines at line 1294 that show this same allocation:

    !   write(*,*) 'maxnumericargsfound=',maxnumericargsfound
    !   write(*,*) 'maxalphaargsfound=',maxalphaargsfound
    !   IF (.not. ALLOCATED(LineItem%Numbers)) ALLOCATE (LineItem%Numbers(MaxNumericArgsFound+20))
    !   IF (.not. ALLOCATED(LineItem%NumBlank)) ALLOCATE (LineItem%NumBlank(MaxNumericArgsFound+20))
    !   IF (.not. ALLOCATED(LineItem%Alphas)) ALLOCATE (LineItem%Alphas(MaxAlphaArgsFound+20))
    !   IF (.not. ALLOCATED(LineItem%AlphBlank)) ALLOCATE (LineItem%AlphBlank(MaxAlphaArgsFound+20))

**Comment by rraustad** [comment]
Diff's on Linux? The transition program doesn't even come into play here does it?

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Calling this good.  The diffs *must* have been from the branch being out of date or something else.  This change has nothing to do with diffs.  The code change makes sense, and although I don't understand why the +20 was there originally, this transitions the defect file successfully, and that's a good test that it's working properly.  I ran the transitioned file and the input is processed successfully.  Done.

**Comment by mjwitte** [comment]
Just to be sure, it would be good to run a full set of v8.7 test files through this to make sure they all work.  But, probably none of the test files trip the extensible part of transition.

**Comment by Myoldmopar** [comment]
I agree on both points. There's probably nothing in our test file arsenal that hits that. And if it can transition that massive object and entire file successfully, that's about as good of a test as we have.  But we can certainly run a few simpler cases to feel better about it I suppose.

**Pull request merged**